anchor/network/src/handshake/mod.rs

```diff
@@ -31,226 +26,148 @@ pub enum Error {
 
 /// Event emitted on handshake completion or failure.
 #[derive(Debug)]
-pub enum Event {
-    Completed {
-        peer_id: PeerId,
-        their_info: NodeInfo,
-    },
-    Failed {
-        peer_id: PeerId,
-        error: Error,
-    },
+pub struct Completed {
```

Comment:
nit: we could probably comment these structures and functions in this module, for better understanding